src/lib/statistics/statistics_objects/equal_distinct_count_histogram.cpp

```diff
@@ -49,11 +50,36 @@ template <typename T>
 std::vector<std::pair<T, HistogramCountType>> value_distribution_from_column(const Table& table,
                                                                              const ColumnID column_id,
                                                                              const HistogramDomain<T>& domain) {
-  // TODO(anybody) If you want to look into performance, this would probably benefit greatly from monotonic buffer
-  //               resources.
-  ValueDistributionMap<T> value_distribution_map;
-
+  // Since this function can be the main bottleneck for loading data in the benchmarks, we put some effort in
+  // optimizing it. For once, we try to get the distinct count in case we have dictionary-encoded data in place. This
+  // allows us to pre-size the robin_map which helps a lot with performance (no resizes for large maps) and memory
+  // consumption (as we can avoid pre-sizes all maps with the table size, which is often by far too large).
+
+  // Use a defensive default in case we cannot get a better estimate (i.e., no dictionary-encoded data).
+  auto estimated_distinct_value_count = std::min(table.row_count(), uint64_t{1'024});
+  auto max_load_factor = std::optional<float>{};
   const auto chunk_count = table.chunk_count();
+
+  // Meta tables do not have chunks, we thus need to check for having a chunk count larger zero.
+  if (chunk_count > 0 && table.get_chunk(ChunkID{0})) {
+    // Check if first chunk is dictionary-encoded.
+    if (const auto* dictionary_segment =
+            dynamic_cast<const BaseDictionarySegment*>(&*table.get_chunk(ChunkID{0})->get_segment(column_id))) {
+      estimated_distinct_value_count = std::max(
+          table.row_count(),
+          static_cast<uint64_t>(1.1 * static_cast<double>(chunk_count * dictionary_segment->unique_values_count())));
+      // For rather accurate distinct count estimates, the max_load_factor is upped as we are rather sure than no
+      // resize will happen.
+      max_load_factor = 0.9f;
```

Comment:
upped as in increased? In my opinion it needs to have a value to be increased but the optional should not have a value here? Also, I don't understand the argument made in the comment, is the above procedure called `rather accurate`?

Reply:
than -> that

Reply:
Using the number of only the first segment's distinct values and multiplying it by the number of chunks assumes a uniform value distribution and that all chunks are existing, doesn't it?

What is the rationale for the factor `1.1`?

Using `MAXIMUM_MAX_LOAD_FACTOR` instead of 0.9f might be more understandable for the reader? To me, 0.9f looks like a magic number.